op-chain-ops/genesis/setters.go

```diff
@@ -89,3 +90,11 @@ func setupPredeploy(db vm.StateDB, deployResults immutables.DeploymentResults, s
 
 	return nil
 }
+
+// setup4788Contract creates the EIP-4788 beacon-block-roots contract, part of the Ecotone upgrade.
+func setup4788Contract(db vm.StateDB) error {
+	db.CreateAccount(predeploys.EIP4788ContractAddr)
+	db.SetCode(predeploys.EIP4788ContractAddr, predeploys.EIP4788ContractCode)
+	db.SetNonce(predeploys.EIP4788ContractAddr, 1) // After contract deployment, the resulting contract has nonce=1, see EIP-158
```

Comment:
should this be EIP-161 instead of 158?

Reply:
EIP-158, at least that's what the geth source mentions about the nonce increase in the contract-creation code-path: https://github.com/ethereum/go-ethereum/blob/c66ca8bf7a8c63ae54e44f4566e206cd1a4fa204/core/vm/evm.go#L450